test/e2e/suite_test.go

```diff
@@ -386,10 +388,13 @@ func beforeTest(t *testing.T, f *framework.E2eFramework, opts *e2eTestOpts) erro
 	}
 
 	if opts.deployTraefik {
-		if err := f.DeployTraefik(t, opts.operatorNamespace); err != nil {
-			t.Logf("failed to deploy Traefik")
-			return err
-		}
+		retry.Do(func() error {
```

Comment:
What's the benefit over assert.Eventually or whatever we otherwise use in the tests? Seems bad to introduce yet another framework for doing the same thing, only increases maintenance burden.

Reply:
Using `Eventually` was my first idea, but I seem to recall wouldn't work with retries on Stargate, as there are some places where we need to get something in return of the call.
Let me check back all the places where I'm using `retry.Do` to see if it's legit or if `Eventually` can be used everywhere.
I agree on the principle that introducing a new dependency should only be done if the ones we have don't provide the functionality we're after.

Reply:
After checking again, you're totally right. `require.Eventually` will do the job. I'll switch to this and remove the `retry` dependency.